core-api/src/main/java/com/optimizely/ab/optimizelyconfig/OptimizelyConfigService.java

```diff
@@ -215,4 +292,60 @@ String getExperimentFeatureKey(String experimentId) {
 
         return featureVariableKeyMap;
     }
+
+    @VisibleForTesting
+    List<OptimizelyAudience> getAudiencesList(List<Audience> typedAudiences, List<Audience> audiences) {
+        /*
+         * This method merges typedAudiences with audiences from the Project
+         * config. Precedence is given to typedAudiences over audiences.
+         *
+         * Returns:
+         *   A new list with the merged audiences as OptimizelyAudience objects.
+         * */
+        List<OptimizelyAudience> audiencesList = new ArrayList<>();
+        Map<String, String> idLookupMap = new HashMap<>();
+        if (typedAudiences != null) {
+            for (Audience audience : typedAudiences) {
+                OptimizelyAudience optimizelyAudience = new OptimizelyAudience(
+                    audience.getId(),
+                    audience.getName(),
+                    audience.getConditions().toString()
+                );
+                audiencesList.add(optimizelyAudience);
+                idLookupMap.put(audience.getId(), audience.getId());
+            }
+        }
+
+        if (audiences != null) {
+            for (Audience audience : audiences) {
+                if (!idLookupMap.containsKey(audience.getId()) && audience.getId() != "$opt_dummy_audience") {
+                    OptimizelyAudience optimizelyAudience = new OptimizelyAudience(
+                        audience.getId(),
+                        audience.getName(),
+                        audience.getConditions().toString()
+                    );
+                    audiencesList.add(optimizelyAudience);
+                }
+            }
+        }
+        
+        return audiencesList;
+    }
+
+    @VisibleForTesting
+    Map<String, String> getAudiencesMap(List<OptimizelyAudience> optimizelyAudiences) {
+        Map<String, String> audiencesMap = new HashMap<>();
+
+        // Build audienceMap as [id:name]
+        if (optimizelyAudiences != null) {
```

Comment:
Do you need this null check? Should you return null if you get one here?

Reply:
Same comment here as above with the addition of looking into if I should return null rather than the empty map.